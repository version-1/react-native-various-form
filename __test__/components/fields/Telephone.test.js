import React, { Component } from 'react'
import { Text, StyleSheet } from 'react-native'
import renderer from 'react-test-renderer'
import { Telephone } from 'react-native-various-form'

describe('Telephone', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Telephone value="000-0000-0000" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('placeholder is changed as passed', () => {
    const tree = renderer
      .create(<Telephone value="000-0000-0000" placeholder="000-0000-0000" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('delimiter is changed as passed', () => {
    const tree = renderer
      .create(<Telephone value="000 0000 0000" delimiter=" " />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
