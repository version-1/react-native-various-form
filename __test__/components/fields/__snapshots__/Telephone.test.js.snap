// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Telephone delimiter is changed as passed 1`] = `
<TextInput
  allowFontScaling={true}
  delimiter=" "
  keyboardType="phone-pad"
  maxLength={13}
  onChangeText={[Function]}
  onKeyPress={[Function]}
  placeholder="000-0000-0000"
  value="000 0000 0000"
/>
`;

exports[`Telephone placeholder is changed as passed 1`] = `
<TextInput
  allowFontScaling={true}
  keyboardType="phone-pad"
  maxLength={13}
  onChangeText={[Function]}
  onKeyPress={[Function]}
  placeholder="000-0000-0000"
  value="000-0000-0000"
/>
`;

exports[`Telephone renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  keyboardType="phone-pad"
  maxLength={13}
  onChangeText={[Function]}
  onKeyPress={[Function]}
  placeholder="000-0000-0000"
  value="000-0000-0000"
/>
`;
